### PR TITLE
feat(std/testing): Add assertExists assertion

### PR DESCRIPTION
**Pull request by getspooky**

### CODE REVIEW

File: std/testing/asserts.ts [hunk 1]
```diff
@@ -329,6 +329,23 @@ export function assertNotStrictEquals(
...
 }
 
+/**
+ * Make an assertion that actual is not null or undefined. If not
+ * then thrown.
+ */
+export function assertAnything(
+  actual: unknown,
+  msg?: string,
+): void {
```
Comment (bartlomieju) [inline std/testing/asserts.ts:339]: It seems you can already achieve the same assertion using just `assert()`. Is there some edge case not covered by it?

Comment (kitsonk) [reply std/testing/asserts.ts:339]: `assert(false)` would throw, while `assertAnything(false)` would not.

File: std/testing/asserts_test.ts [hunk 2]
```diff
@@ -159,6 +160,19 @@ Deno.test("testingNotEquals", function (): void {
   assertEquals(didThrow, true);
 });
 
+Deno.test("testingAssertAnything", function (): void {
+  assertAnything("Denosaurus");
+  let didThrow;
+  try {
```
Comment (kitsonk) [inline std/testing/asserts_test.ts:166]: We should test all the [falsey](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) values here.

**Review by kitsonk — COMMENTED** [review]
I would prefer `assertExists()`.

Prior art: https://www.chaijs.com/api/assert/#method_exists

File: std/testing/asserts.ts (hunk 1 above)
Comment (caspervonb) [reply std/testing/asserts.ts:339]: I prefer explicit assertions, if you're checking that something is not null and not undefined there's probably a very specific reason for that.

```typescript
assert(value !== undefined && value !== null);
```

File: std/testing/asserts_test.ts (hunk 2 above)
Comment (getspooky) [reply std/testing/asserts_test.ts:166]: @kitsonk  I think assertExists it's a good idea

**Comment by bartlomieju** [comment]
Looks good to me, waiting on @kitsonk's approval

File: std/testing/asserts_test.ts [hunk 3]
```diff
@@ -159,6 +160,19 @@ Deno.test("testingNotEquals", function (): void {
   assertEquals(didThrow, true);
 });
 
+Deno.test("testingAssertExists", function (): void {
+  assertExists("Denosaurus");
```
Comment (kitsonk) [inline std/testing/asserts_test.ts:164]: Still think we need to have proper coverage for this...  Especially with ["falsey" values](https://developer.mozilla.org/en-US/docs/Glossary/Falsy).

```ts
assertExists(false);
assertExists(0);
assertExists("");
assertExists(-0);
assertExists(0n);
assertExists(NaN);
```

File: std/testing/asserts_test.ts [hunk 4]
```diff
@@ -159,6 +160,19 @@ Deno.test("testingNotEquals", function (): void {
...
+  let didThrow;
+  try {
+    assertExists(undefined);
```
Comment (kitsonk) [inline std/testing/asserts_test.ts:167]: We should test `null` and `void 0` in the same way.

**Review by kitsonk — DISMISSED** [review]
Really want to see full coverage in the tests, as we want to really want to be able to trust that something like an assertion will work as designed.

File: std/testing/asserts_test.ts (hunk 4 above)
Comment (getspooky) [reply std/testing/asserts_test.ts:174]: @kitsonk I added some falsy test

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you @getspooky!

**Pull request merged**